web/page0/src/app/runtime-card/runtime-card.component.ts

```diff
@@ -11,20 +15,61 @@ import { RuntimeService } from '../runtime.service';
 export class RuntimeCardComponent {
   @Input() runtime: Runtime | null = null;
 
-  // TODO: chain zone - host - group list APIs
+  hosts$ = new BehaviorSubject<Host[]>([]);
+
+  private ngUnsubscribe = new Subject<void>();
+
+  constructor(
+    private router: Router,
+    private runtimeService: RuntimeService,
+    private hostService: HostService,
+    private snackBar: MatSnackBar
+  ) {}
+
+  ngOnInit() {
+    if (!this.runtime) {
+      return;
+    }
+
+    this.hostService
+      .getHosts(this.runtime.alias)
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((hosts) => this.hosts$.next(hosts));
+  }
+
+  ngOnDestroy() {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
 
-  constructor(private router: Router, private runtimeService: RuntimeService) {}
   onClickAddHost() {
     this.router.navigate(['/create-host'], {
       queryParams: { runtime: this.runtime?.alias },
     });
   }
 
-  onClickDelete(alias: string | undefined) {
+  onClickUnregister(alias: string | undefined) {
     if (!alias) {
       return;
     }
 
     this.runtimeService.unregisterRuntime(alias);
   }
+
+  onClickDeleteHost(host: Host) {
+    this.snackBar.open(`Start to delete host ${host.name} (url: ${host.url})`);
+    this.hostService
+      .deleteHost(host.url)
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe({
+        next: () => {
+          this.snackBar.dismiss();
+        },
+        error: (error) => {
+          this.snackBar.open(
+            `Failed to delete host ${host.url} (error: ${error.message})`
+          );
```

Comment:
How about adding dismissal for snackbar?

https://material.angular.io/components/snack-bar/overview#dismissal

Reply:
Done